boundary_layer_default_plugin/preprocessors.py

```diff
@@ -42,6 +42,42 @@ def process_arg(self, arg, node, raw_args):
 
         return date
 
+class BuildKubernetesSchema(StrictSchema):
+    class_name = ma.fields.String(required=True)
+
+    @ma.validates_schema
+    def check_valid_class(self, data):
+        ALLOWED_CLASS = ['airflow.contrib.kubernetes.volume.Volume', 'airflow.contrib.kubernetes.volume_mount.VolumeMount', 'airflow.contrib.kubernetes.secret.Secret', 'airflow.contrib.kubernetes.pod.Resources']
+        if data.get('class_name') not in ALLOWED_CLASS:
+            raise ma.ValidationError(
+                '`class_name` must be one of `{}`'.format(
+                    '`, `'.join(ALLOWED_CLASS)),
+                ['class_name'])
+
+        
+class KubernetesPrep(PropertyPreprocessor):
+    type = "kubernetes_prep"
+
+    properties_schema_cls = BuildKubernetesSchema
+    def imports(self):
+        return {'modules': ['.'.join(self.properties['class_name'].split('.')[:-1])]}
+
+    def process_arg(self, arg, node, raw_args):
+        kube_objects = None
+        try:
+            if isinstance(arg, list):
+                kube_objects = [GenericShell(self.properties['class_name'], a) for a in arg]
```

Comment:
This is an interesting way to deal with the need to embed constructed classes in the DAG definition.  I don't think we've faced this problem before, but I could imagine it having more general utility beyond Kubernetes, so thanks for establishing a pattern for it!